### ISSUE TITLE
Idle kernel threads missing on Linux processes

### ISSUE THREAD

**Issue by damluk** [munin-monitoring/munin#1144]
Ubuntu disco man ps:
```
PROCESS STATE CODES
...
               D    uninterruptible sleep (usually IO)
               I    Idle kernel thread
```
This idle state is currently not graphed under Linux, and it is many of them.

**Comment by sumpfralle** [comment]
In which specific plugin did you miss that information?
Maybe you want to prepare a pull request? :)

**Comment by damluk** [comment]
In `plugins/node.d/processes`. Can do a merge request later.

**Comment by sumpfralle** [comment]
Perfect!

**Comment by damluk** [comment]
Actually it is pretty hard to be as generic as possible. According to procps changelog, Kernel 4.2 introduced the new state, so I will need to compare kernel versions. When doing that, I need to remain POSIX compliant. I won't be able to provide a generic versionstring comparison function.

**Comment by sumpfralle** [comment]
How about something like the following?

```shell
[ "$min_version" = "$(printf '%s\n' "$min_version" "$current_version" | sort --version-sort)" ]
```
(or a more readable variation)

**Comment by sumpfralle** [comment]
How about keeping it simple and just add the "I" indicator to the linux parser?
Older systems will see this value as zero.
Newer systems will get a real value.

What do you think?

**Comment by sumpfralle** [comment]
#1157 was merged into `stable-2.0`
